core/src/com/unciv/logic/automation/WorkerAutomation.kt

```diff
@@ -52,7 +61,7 @@ class WorkerAutomation(val unit: MapUnit) {
         val targetRoad = unit.civInfo.tech.getBestRoadAvailable()
 
         val citiesThatNeedConnecting = unit.civInfo.cities
-                .filter { it.population.population>3 && !it.isCapital()
+                .filter { it.population.population>3 && !it.isCapital() && !it.isBeingRazed //City being razed should not be connected.
```

Comment:
Why not?

Reply:
This is for player's connivence. When you capture a city on a bad location and decide to demolish it, you won't wish workers build many road tiles for it. Will this bring in any problem?

Reply:
I see, good point!